### ISSUE TITLE
Confirmable documentation outdated

### ISSUE THREAD

**Issue by jamie** [plataformatec/devise#3508]
The documentation for `Confirmable#confirmation_period_expired?` is outdated, and the examples listed [here](https://github.com/plataformatec/devise/blob/master/lib/devise/models/confirmable.rb#L194-L201) should say `allow_unconfirmed_access_for` instead of `confirm_within`, as the latter was removed in 2.0.

**Comment by nashby** [comment]
Well it's kinda messy. As I can see it was deprecated [here](https://github.com/plataformatec/devise/commit/930b324c15ab54f3682cb6d78405e46b24d25ffb) but then is was brought back in [this](https://github.com/plataformatec/devise/pull/1975) pull request. So as I can see the documentation is ok. @josevalim am I right?

**Comment by jamie** [comment]
I'm pretty sure the documentation is correct, it just references the wrong method name. And has for a good while.

**Comment by jlecour** [comment]
**If I understand correctly**, `allow_unconfirmed_access_for` and `confirm_within` are 2 different things.

* `allow_unconfirmed_access_for` allows to verify the account later.
* `confirm_within` sets a timeout for the validity of the token.


For example, if I set `allow_unconfirmed_access_for` to `30.days`, and no value for `confirm_within` I'll have to confirm my account within 30 days, but I do it 3 months later, the link is still valid.

If I set `confirm_within` to 1 hour, no matter how long the verification time window is, after 1 hour, the link in the email becomes invalid.

This is how I understand it, but **i might be wrong**.
